pkg/appendable/appendable.go

```diff
@@ -134,17 +134,19 @@ func (m *FileMeta) UnmarshalBinary(buf []byte) error {
 }
 
 type IndexMeta struct {
-	FieldName string
-	FieldType FieldType
-	Width     uint16
+	FieldName   string
+	FieldType   FieldType
+	Width       uint16
+	TotalLength uint64
```

Comment:
Maybe name this something like `TotalValueLength`? It's not clear what length it's referring to. A comment would also be good.